sale_order_type/models/sale_order.py

```diff
@@ -62,8 +62,8 @@ def create(self, vals):
     @api.multi
     def write(self, vals):
         res = super(SaleOrder, self).write(vals)
-        if vals.get("type_id") and (self.state == "draft" or self.state == "sent"):
-            for order in self:
+        for order in self:
+            if vals.get("type_id") and (order.state in ["draft", "sent"]):
```

Comment:
Please evaluate this and if there's a sequence before calling super, and also checking if the record was previously with the same type, and modify dictionary for calling super for only write once. A quick code snippet would be:
```python
if vals.get("type_id"):
    sale_type = self.env["sale.order.type"].browse(vals["type_id"])
    if sale_type.sequence_id:
        for record in self:
            if record.state in {"draft", "sent"} and record.type_id != sale_type:
                new_vals = vals.copy()
                new_vals["name"] = sale_type.sequence_id.next_by_id()
                super(SaleOrder, record).write(new_vals)
            else:
                super(SaleOrder, record).write(vals)
```

Reply:
@pedrobaeza thanks for the hint. just a question `{"draft", "sent"}` is correct? i'd expect `["draft", "sent"]`

Reply:
Yes, it's correct. It's the Python definition of a set, that improves better with `in` operator, although the other works as well

Reply:
k, thanks!